libs/brig-types/src/Brig/Types/Common.hs

```diff
@@ -214,33 +211,16 @@ ssoIdentity _ = Nothing
 
 -- | TODO: once we have @/libs/spar-types@ for the wire-sso-sp-server called spar, this type should
 -- move there.
-data UserSSOId = UserSSOId { userSSOIdTenant :: UUID, userSSOIdSubject :: UUID }
-    deriving (Eq)
-
-instance Show UserSSOId where
-  show = show . userSSOIdToText
-
-instance IsString UserSSOId where
-  fromString = either (error . ("fromString @UserSSOId: " <>)) id . userSSOIdFromText . Text.pack
-
-userSSOIdFromText :: Text -> Either String UserSSOId
-userSSOIdFromText raw = do
-    ix <- maybe (Left "UserSSOId: ':'") pure $ Text.findIndex (== ':') raw
-    case Text.splitAt ix raw & _2 %~ Text.splitAt 1 of
-        (tenant, (":", subject)) -> case (fromText tenant, fromText subject) of
-            (Nothing, _)      -> Left "UserSSOId: tenant"
-            (Just _, Nothing) -> Left "UserSSOId: subject"
-            (Just t, Just s)  -> pure $ UserSSOId t s
-        bad -> Left $ "UserSSOId: impossible: " <> show bad
-
-userSSOIdToText :: UserSSOId -> Text
-userSSOIdToText (UserSSOId tenant subject) = toText tenant <> ":" <> toText subject
+data UserSSOId = UserSSOId { userSSOIdTenant :: Text, userSSOIdSubject :: Text }
```

Comment:
So `Tenant` and `Subject` used to be `UUID`s... is that no longer true?

Reply:
correct.  tenant is a URL (except for in the centrify example, there it is just a name, which violates the specs).  subject is a NameID, which may be a URI or a UUID or something else, with 3 optional qualifying namespace-URIs that all have their own meaning.

this is not great, but UUID didn't work out with the sample xml snippets i've found.

Reply:
- Uniqueness of subject inside the tenant namespace is the IdP's responsibility.
- Uniqueness of the tenant is our responsibility.  Currently, we only have a config file and need to verify this manually.  As soon as we are going to start storing the IdP information in cql, we will need to make sure the newly created IdPs do not delete old ones.  If anybody tries, they will get an error.

is that a good plan?